### DISCUSSION TITLE
24 volt AC to 24 volt DC Rectifier Recommendations Sought

### DISCUSSION THREAD

**Discussion by FlaMike** [tagyoureit/nodejs-poolController#837] [Q&A]
I am in the early planning stages of building a Nixie and taking advantage of the current ET4's infrastructure (transformers, relays, etc).  In particular, I am inspired by codeslinger69's project https://github.com/tagyoureit/nodejs-poolController/discussions/801 and hope to follow his well-documented thread and the wisdom gained through his experiences.  Moving forward, I will start a new thread to share my experience and, hopefully, successes.

I know the system will need both 24 V AC and 24 V DC power available.   It's been suggested the ET has 24V AC available and using a rectifier may suffice to meet the 24 V DC requirements.  As I am planning out the project, I am trying to identify an appropriate and robust rectifier to meet this need.  Doing an initial search I found this regulator board that may or may be up to the job.  

https://www.amazon.com/UMLIFE-Converter-2-5-35V-Regulator-Adjustable/dp/B094ZTG5S8/ref=sr_1_3?keywords=24%2Bvolt%2Bac%2Bto%2B24%2Bvolt%2Bdc%2Brectifier&sr=8-3&th=1

If you have experience with using a rectifier for the 24 V AC-DC conversion, can you kindly share your experience and recommendations?

Thank you for you help!!

**Comment by rock-crusher** [comment]
Well, if you’re going to do a full-on cannibalization of your ET, I suppose you could just use the ET circuit board along with the x-former for both the 24vac & 24vdc.

**Reply by CodeSlinger69** [reply]
My ET doesn't have a built in VDC transformer. I did salvage and reuse the ET's 24VAC transformer, but then I bought [this one](https://www.amazon.com/gp/product/B00R5CI1W0/) for DC.

There may very well be DC via a rectifier on the ET board somewhere, but I didn't probe for it. Just easier to spend the $35 for this piece, for me at least.

**Reply by FlaMike** [reply]
I'll have to poke around a bit to see if there's a ready 24V DC supply available in the existing controller.  Currently waiting for the heat index to drop below 100 for a little while :-)

If not, I agree the Meanwell transformer is a good option, especially if the Pi stack is being mounted on DIN rails.  Probably much more reliable than the regulator board, too.

**Reply by rstrouse** [reply]
You can use a simple bridge rectifier for this.  No need for a buck.

**Reply by FlaMike** [reply]
Thank you.  Are there any you have used and recommend?  I'm driving a bit ahead of my headlights & would like to make sure I'm getting something appropriate.

**Reply by rock-crusher** [reply]
No need to probe for the 24vdc. Simply turn the auxiliary relays on manually and the relay terminals on the circuit board will be energized with the 24vdc.
Hey, I know it may be a little unorthodox, but we’re talking cannibalization here and this will provide you with the 24vdc you need for controlling your 3hp relays… just saying

**Reply by FlaMike** [reply]
I love it!  EZ PZ & even I think I can handle it :-)  Thanks for the suggestion!

**Reply by CodeSlinger69** [reply]
I also had a goal to pull out my ET board/display, and maybe use that top space in the box for something else. I don't think I'll end up using that space in the box for anything, as I'll be moving the Nixie equipment over to another box eventually. But that was the original thinking. If you do pull the board, then obv. you'll need a separate AC-DC transformer.

**Reply by FlaMike** [reply]
Good point.  I was thinking about  that as well.  I may leave the ET motherboard in the box so it's  there  & waiting  for the (next?) owner, but I do plan to bypass it entirely once the installation is  complete.  If the ET has a rectifier bridge on it, then you're correct it won't be available to use & I will have to either install a bridge or an additional transformer.  Huge thunderstorm here currently, so there's no way I can even confirm visually at what's  going  on in the box.  Who  knows, maybe I can find a 24V DC source hidden somewhere that's off the motherboard?

Still hoping someone can recommend an available rectifier bridge to purchase.  If so, I can place it in the secondary (low voltage Pi stack) box & skip getting the Meanwell unit.

In the interim this portion of the planning stage is on hold.  Of course, plenty more to cogitate & consider...

**Comment by FlaMike** [comment]
On a slightly different aspect of njs-PC...  In the Wiki, it's recommended that installing a remote desktop app may be beneficial for managing the Pi.  While I do not doubt there is some value to its value to many folks, I am wondering if it may be "better" from a Pi resource utilization standpoint to dispense with a GUI in total and just install the "Lite" version of the OS, without any desktop environment.  It appears that running raspi-config via ssh provides a straightforward  mechanism to do things like enabling i2c (which can also be accomplished via the command line).

I do not doubt the experience & wisdom of the author(s) of the Wiki, as they are far more experienced and knowledgeable than am I.  Nevertheless, when running a small computer like a Pi, GUIs and RDP connections may tax the little fellow more than is worthwhile.

I should also add that, in my case, I am using a Pi 3B+, not a Pi 4 with 8 GB, so the capability of the Pi is even more limited that the current recommended approach suggests.

Please feel free to opine as to the long-term value of having a GUI and RDP on the Pi.  I would not be surprised if there are issues of which I am ignorant that suggest their inclusion is valuable and I should just go forth and install, per the Wiki.

Thanks for your help & suggestions!

**Reply by CodeSlinger69** [reply]
[I have a RPi 3b](https://github.com/tagyoureit/nodejs-poolController/discussions/801#discussioncomment-6287923), and it has plenty of power to run njsPC and a desktop environment. njsPC is not at all demanding, I see 3-5% utilization at most.

**Reply by FlaMike** [reply]
Thanks for your quick & helpful response.  I guess it's a non-issue which is great news.  The old coot in me said minimize the software footprint unless stuff is really needed (from the days when bits were big & memories & CPUs were "small" in capability).  Good to know even a middle-aged Pi has plenty of horsepower for a "full"  software suite.

I have a spare SSD & SATA to USB box, so I am using it instead  of a micro SD.  Hoping it's more reliable as well as having lots of storage--for what,  I don't know.

My progress report to date is I loaded up the Pi with the complete (with GUI) OS, installed the prerequisite software, added the Pool Controller, REM, and Dashboard, PM2, logrotate, and  remote desktop.  I enabled I2C even though I have not even ordered the two needed hats yet.  Everything seems OK so far, although the Pi is complaining about  an underpowered power supply, 2500 mA.  I can't run Pi Configuration program through the GUI possibly as a result of low power, so I enabled I2C via the command line.  I can also get to the "interfaces" section of raspi-config via SSH, so it's all good as far as I can tell at the moment.  For the record, the aforementioned power supply is temporary for Pi on my desk use, not "real" pool controller work.  I can also plug a USB cable into a wall outlet with USB ports if I need more power while at my desk.

Don't think there's  too  much else I can do at this time until I get some parts.

One other basic & probably stupid question comes to mind...  What are folks using to make connections for wiring?  Wire nuts or something more elegant?  Russ talks about using JST-XH connectors for the relay diodes, do folks use them throughout the controller?  I assume in my implementation there will be at least a handful or two of wires that will be connecting to existing wires because I'm keeping the existing Pentair & high voltage stuff in the existing box & putting the Pi stack & any related low voltage goodies in a box yet to be purchased.  I have a ferule kit for the low voltage terminators at the hats/relays.

Other than getting the hats, a box to put it in, some Din rails/mounts/terminal blocks, etc. what am I missing/forgetting at this stage?  

Of course, more study of @CodeSlinger69's #801 thread and @tagyoureit and @rstrouse many Wiki & thread posts continue to limit as many oopses as possible :-)

Thanks y'all!

**Reply by CodeSlinger69** [reply]
Ya, you're pretty much asking questions now that are answered in 801 and the wiki.

**Reply by FlaMike** [reply]
OK, thanks.  I shall re-direct my full attention to them & go from there.
